src/plugins/data/common/search/aggs/buckets/lib/time_buckets/time_buckets.ts

```diff
@@ -172,38 +177,31 @@ export class TimeBuckets {
    * @param {object|string|moment.duration} input - see desc
    */
   setInterval(input: null | string | Record<string, any>) {
-    let interval = input;
-
-    // selection object -> val
-    if (isObject(input) && !moment.isDuration(input)) {
-      interval = input.val;
-    }
+    this._i = undefined;
+    this._originalInterval = null;
 
-    if (!interval || interval === autoInterval) {
+    if (!input || input === autoInterval) {
       this._i = autoInterval;
       return;
     }
 
-    if (isString(interval)) {
-      input = interval;
-
-      // Preserve the original units because they're lost when the interval is converted to a
-      // moment duration object.
-      this._originalInterval = input;
-
-      interval = parseInterval(interval);
-      if (interval === null || +interval === 0) {
-        interval = null;
+    if (moment.isDuration(input)) {
+      this._i = input;
+    } else if (isObject(input)) {
+      // selection object -> val
+      this._i = input.val;
```

Comment:
we are still not handling the case where input is an object and the val property of it is a string `{ val: 'timestring' }` which we did before. i am not sure its needed but unless this was intentional we should probably add it back.

Reply:
@ppisljar thank you. this  method looks difficult and do nothing, I think we can simplify it. I have updated the PR